doc: move wiki to Doxygen documentation (second attempt)

jia200x opened the pull request:
<!--
The RIOT community cares a lot about code quality.
Therefore, before describing what your contribution is about, we would like
you to make sure that your modifications are compliant with the RIOT
coding conventions, see https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions.
-->

### Contribution description
This is a second attempt of moving Board specifications from Wiki to Doxygen. Replaces the work in #8508 .

Wiki entries were copied "as it" and referenced to the corresponding group, [with the same format as existing Doxygen entries like Arduino MKRFOX1200](http://doc.riot-os.org/group__boards__arduino-mkrfox1200.html)
As said in the previous PR, the motivation is to help developers keeping an up-to-date End User documentation for each board without adding too much overhead, and help the End User to find all the documentation in the same place. 

- [x] deduplicate `@defgroup` and `@brief`
- [x] remove tabs
- [x] split lines to fit 80 chars

NB tables doesn't work if splitted => NR > 80. Also, this PR doesn't intend to update information for each entry.

### Some other issues addressed by this PR
* Fix mulle `todo` block

### Entries with broken imgs (addressed in further PRs)
- arduino-mega2560
- samd21-xpro
- saml21-xpro
- stm32f3discovery
- z1

### Following steps (PRs coming soon)
* Add "Family" documentation (ARM, MSP430, etc).
* Unify the current board documentation structure, so all boards have the same content (Hardware Description, Pin Mapping, Toolchain, etc).
### Issues/PRs references
#8508
<!--
Examples: Fixes #1234. See also #5678. Depends on PR #9876.

Please use keywords (e.g., fixes, resolve) with the links to the issues you
resolved, this way they will be automatically closed when your pull request
is merged. See https://help.github.com/articles/closing-issues-using-keywords/.
-->

aabadie commented:
Very nice ! I like it.

Unfortunately, with this PR, I think each board group definition (using `@defgroup`) is duplicated between the doc.txt and the board.h files: the board.h should be updated as well... for each board !

jia200x commented:
@aabadie I think I found a way to automate that. I will try :)

kaspar030 commented on boards/airfy-beacon/doc.txt:
```diff
@@ -0,0 +1,209 @@
+ * 
+ * ![airfy-beacon](https://raw.githubusercontent.com/wiki/RIOT-OS/RIOT/images/airfy-beacon.jpg)
+ * 
+ * | MCU 		| NRF51822QFAA 		|
```
should use spaces instead of tabs...

aabadie replied:
The original file in the wiki already contains the tabs, maybe the automatic script could also handle that. What do you think @jia200x ?

jia200x replied:
Bash magic may do the trick :)

gebart replied:
`expand` may help here

aabadie commented on boards/airfy-beacon/doc.txt:
```diff
@@ -0,0 +1,209 @@
+ * ## Overview
+ * 
+ * The Airfy Beacon is utilizing a Nordics NRF51822QFAA SoC.
+ * The SoC features 16Kb of RAM, 256Kb of flash ROM and comes on top of the usual micro-controller peripherals with a 2.4GHz radio that supports both Nordics proprietary ShockBurst as well as Bluetooth Low Energy (BLE).
```
It would be great to split the line so that they fit in the 80 characters maximum length. Do you think it's possible to adapt your script for this ?

jia200x replied:
yes, I think so

jia200x commented:
@aabadie thanks a lot for reviewing!

jia200x commented:
@aabadie could you help me with the review of this?

aabadie commented on boards/acd52832/doc.txt:
```diff
@@ -0,0 +1 @@
+ */
```
Looks like a bug: your script is creating a doc.txt when the board is not documented in the code base and not present in the wiki. I saw a few others like this.

jia200x replied:
Yes, it's a bug. I will fix it when I'm back

jia200x replied:
fixed

aabadie commented on boards/iotlab-a8-m3/doc.txt:
```diff
@@ -0,0 +1,4 @@
+/**
+ * @ingroup     boards_iotlab-m3
```
There is a problem here, the group should be `boards_iotlaba-a8-m3`. This is because the initial group of the A8 was invalid.

aabadie commented on boards/mulle/include/board.h:
```diff
@@ -104,8 +102,10 @@ void board_init(void);
- *
- * @todo Work around missing RESET pin on Mulle v0.6x
+ * @{
+ */
```
Don't need to close the doxygen block here

jia200x replied:
I will try again, but I strange error on `make doc` without splitting this block. I will paste the results here

jia200x replied:
this is my output when I don't split the block:

```
( cat riot.doxyfile ; echo "GENERATE_HTML = yes" ) | doxygen -
todo:17: warning: Unexpected character `b'
todo:17: warning: Unexpected character `o'
todo:17: warning: Unexpected character `a'
todo:17: warning: Unexpected character `r'
todo:17: warning: Unexpected character `d'
todo:17: warning: Unexpected character `s'
todo:17: warning: Unexpected character `/'
todo:17: warning: Unexpected character `m'
todo:17: warning: Unexpected character `u'
todo:17: warning: Unexpected character `l'
todo:17: warning: Unexpected character `l'
todo:17: warning: Unexpected character `e'
todo:17: warning: Unexpected character `/'
todo:17: warning: Unexpected character `i'
todo:17: warning: Unexpected character `n'
todo:17: warning: Unexpected character `c'
todo:17: warning: Unexpected character `l'
todo:17: warning: Unexpected character `u'
todo:17: warning: Unexpected character `d'
todo:17: warning: Unexpected character `e'
todo:17: warning: Unexpected character `/'
sh: latex: no se encontró la orden
error: Problems running latex. Check your installation or look for typos in _formulas.tex and check _formulas.log!
sh: dvips: no se encontró la orden
error: Problems running dvips. Check your installation!
make[1]: se sale del directorio '/home/jialamos/Development/RIOT/doc/doxygen
```

Maybe a Doxygen bug?

aabadie replied:
What about moving the `@todo` before the `@{` ?

jia200x replied:
same results. Does it work for you? Maybe my Doxygen version has a bug

gebart replied:
You can drop the todo. It's not important

aabadie replied:
I also have the issue, but it works with the actual version of your branch. So let's keep it like this

jia200x replied:
or maybe I just drop the `@todo` as @gebart proposed, so the change is minimal

aabadie commented on boards/mulle/include/board.h:
```diff
@@ -104,8 +102,10 @@ void board_init(void);
+ * @{
+ */
+
+ /** @todo Work around missing RESET pin on Mulle v0.6x
```
and repoen it here

aabadie commented on boards/nucleo144-f413/include/periph_conf.h:
```diff
@@ -8,9 +8,7 @@
- * @defgroup    boards_nucleo144-f413 STM32 Nucleo144-F413
- * @ingroup     boards_common_nucleo144
- * @brief       Support for the STM32 Nucleo144-F413
+ * @ingroup     boards__nucleo144-f413common_nucleo144
```
Looks like a bug here

jia200x replied:
thanks for pointing it out

aabadie commented on boards/nucleo144-f429/include/periph_conf.h:
```diff
@@ -7,9 +7,7 @@
- * @defgroup    boards_nucleo144-f429 STM32 Nucleo144-F429
- * @ingroup     boards_common_nucleo144
- * @brief       Support for the STM32 Nucleo144-F429
+ * @ingroup     boards__nucleo144-f429common_nucleo144
```
here as well

aabadie commented on boards/nz32-sc151/include/periph_conf.h:
```diff
@@ -7,7 +7,7 @@
 
 /**
- * @ingroup     nz32-sc151
+ * @ingroup     boards_nz32-sc151
```
Wasn't this fixed already in master ?

jia200x replied:
yes, forgot to rebase

aabadie commented on boards/remote-reva/include/board.h:
```diff
@@ -8,9 +8,7 @@
- * @defgroup    boards_reva RE-Mote Revision A
- * @ingroup     boards
- * @brief       Support for the RE-Mote board Revision A
+ * @ingroup     boards_reva
```
The original group name of this board could have been more specific and then consistent with other `remote` boards: boards_remote-reva

aabadie commented on boards/spark-core/doc.txt:
```diff
@@ -0,0 +1,121 @@
+pause/*2
```
bug ?

jia200x replied:
I was suffering [this bug](https://vi.stackexchange.com/questions/13971/vim-pasting-yank-register-when-opening-a-new-file) in my local vim.

aabadie commented:
@jia200x, I opened #8748 and #8749 for the problem found with doxygen groups.

Otherwise, this PR is getting close from being merged I think. Unfortunately (for me), it will conflict a lot with #8649 #8650 and #8651 since they touch the same parts of the code. But we should merge your PR first because mines are changing the doxygen group names of the boards and I think your automatic script will not work anymore if we merge them first.

jia200x commented:
>@jia200x, I opened #8748 and #8749 for the problem found with doxygen groups. Otherwise, this PR is getting close from being merged I think. Unfortunately (for me), it will conflict a lot with #8649 #8650 and #8651 since they touch the same parts of the code. But we should merge your PR first because mines are changing the doxygen group names of the boards and I think your automatic script will not work anymore if we merge them first.

Great. OK, I will try to fix these issues asap and contribute to merge the other PRs. 

Regarding the changes of ST platforms, it seems the problem would be solved with a small script that moves `@defgroup' and `@brief` some periph_conf.h to doc.txt. Not sure if there are any other files to update.
What do you think?

aabadie commented:
I built locally the documentation to have a look at the result. My first impression is that the board documentation is making a big step forward.

Some problems:
- some table cells are missing sometimes
- the size of some images could be adjusted (maybe in follow-up PRs)
- doxygen groups are invalid for some Nucleo boards

jia200x commented:
Hi @aabadie 

I put some of the problems in the description (broken imgs, etc). IMO should be addressed in further PRs, since at least includes the same information of the wiki (in worst case, it's still equal to what we have now)

> some table cells are missing sometimes

Can you point me to those tables? Did you try the last commit? (I fixed some tables)

> doxygen groups are invalid for some Nucleo boards

Can you point me to those boards? I fixed some in a previous commit.

Again, thanks a lot for reviewing!

aabadie commented:
> Can you point me to those tables?

Arduino Zero, SAMD/L/R21-Xpro, mulle, iotlab-m3, Limifrog v1, nz32-sc151, sodaq-autonomo, nucleo-f207zg, nucleo144-f413, nucleo144-f429, nucleo144-f446, nucleo32-f031, nucleo32-f042, nucleo-f030, nucleo-f070, nucleo-f072, nucleo-f091, nucleo-f103, ... ,nucleo-f446, stm32f3discovery, fox

> Did you try the last commit?

Yes

> Can you point me to those boards?

Nucleo144-F303, Nucleo144-F412, Nucleo144-F722, Nucleo144-F746,  Nucleo144-F767, Nucleo32-F303,  Nucleo32-L031, Nucleo32-L432, Nucleo-L053, Nucleo-L073

So apparently all nucleo without documentation already in the wiki.

This is because in their periph_conf.h file the parent group is `boards_common_nucleo64` and should be group specific to the board.

jia200x commented:
@aabadie I'm currently checking those issues.

jia200x commented:
@aabadie I found an issue in the Arduino Zero doc file at [this line](https://github.com/RIOT-OS/RIOT/pull/8516/files#diff-490584b726c7851a5d2480daf19869b5R34). Regarding the empty cells, this was inherited from the wiki (check [Mulle](https://github.com/RIOT-OS/RIOT/wiki/Board%3A-Mulle)).

I will fix the periph_conf issue.

jia200x commented:
@aabadie I addressed those periph_conf.h files.

aabadie commented:
> I addressed those periph_conf.h files.

I confirm they are fixed

> Regarding the empty cells, this was inherited from the wiki (check Mulle).

Not sure we are talking of the same thing. I'm fine if some information is missing or is not set. The problem is more that the borders of some cells are missing:

![board_bug](https://user-images.githubusercontent.com/1375137/37151081-6a413c82-22d4-11e8-9bae-61f9e751ebde.png)

It looks like a `|` is missing in the table.

jia200x commented:
@aabadie I see the exactly the same table like this:

![](https://gist.githubusercontent.com/jia200x/daac639c022d5b085e0019b63eadbdb0/raw/0ba118ca01ac40efab57c202506e34c3f3932059/pic.png)

What browser are you using? I checked those files and each row is ending with `|`.

aabadie commented:
> What browser are you using?

Firefox 58. By the way, the rendering of the table is not valid: on your table screenshot, some cells should be split (second and third rows for example)

jia200x commented:
> Firefox 58. By the way, the rendering of the table is not valid: on your table screenshot, some cells should be split (second and third rows for example)

Yeah, I see the issue. I will work on it today.

jia200x commented:
@aabadie done. A space character was needed between `|` signs.

jia200x commented:
@aabadie rebased and squashed

aabadie commented:
Apparently, the CI is complaining because of trailing whitespaces.

jia200x commented:
> Apparently, the CI is complaining because of trailing whitespaces.

Addressed

aabadie commented:
I triggered Murdock, just to have his opinion. @kaspar030 what's yours ?

kaspar030 commented on boards/arduino-zero/doc.txt:
```diff
@@ -0,0 +1,91 @@
+/**
```
doesn't doxygen directly support *.md?

aabadie replied:
Apparently not for automatically parsed/generated documentation (doxygen comment, .txt or .doc files). It does support md for explicitly included pages (mainpage.md, etc).

aabadie replied:
And Doxygen supports markdown syntax in comments.

kaspar030 replied:
Actually, adding e.g., "doc.md" to doxygen's FILE_PATTERNS makes it include plain markdown just fine.
It even parses ```@xxx``` doxygen patterns.

See https://gist.github.com/anonymous/2b1611e7e2f751c93c6067435814db46

I suggest changing all those doc.txt into doc.md as done in above gist, and make them proper markdown files (not the C comment stuff). That way our editor also understand the format.

aabadie replied:
Very good to known. This opens interesting perspectives for improving the documentation (not only boards).

jia200x replied:
cool. I'm scripting that. Push commit soon

aabadie replied:
wait, I just tested it locally, enabling it will parse all .md files in the repo. It has some side effects

aabadie replied:
use `doc.md`

jia200x replied:
did it work for you @aabadie?
The files are being indexed, but I have this left menu now:

![](https://gist.githubusercontent.com/jia200x/daac639c022d5b085e0019b63eadbdb0/raw/e3abfc587997d158ceed274f225cf0d78f8cdf61/img.png)

jia200x commented:
There's an issue with .md files (as seen [here](https://stackoverflow.com/questions/3052036/how-to-include-custom-files-in-doxygen/6336368#6336368))

> Just for completeness: there are 3 possible extensions which doxygen treats as additional documentation files: .dox, .txt, and .doc.
> 
> Files which such extension are hidden from the file index. Inside the file you need to put one or more C/C++ style comment blocks.

It seems to be also Doxygen version dependent. Shall we leave the .txt file with C/C++ comments?

kaspar030 commented:
I managed to sort any .md without extra "doc" entry, by just adding "@ingroup" in the first line, and defining the group somewhere else. It still sorts in slightly different (file then gets its own entry depending on first line).

jia200x commented:
@kaspar030 adding a `@ingroup` still adds a "doc" entry here, but hidden under Modules>Boards>Board_name. What Doxygen version are you using?

kaspar030 commented:
> What Doxygen version are you using?

Doxygen 1.8.14. I'm creating a board's group in its board.h, then add "@ingroup xxx" as second line in doc.md. No other doxygen commands in doc.md.

jia200x commented:
> Doxygen 1.8.14. I'm creating a board's group in its board.h, then add "@InGroup xxx" as second line in doc.md. No other doxygen commands in doc.md.

Same version here. I think I get the same results, with the file with its own entry inside the board group.

jia200x commented:
@kaspar030 @aabadie I'm facing to issues with this proposal:
- The Markdown part is added as another entry inside the board, and showed below #defines, etc.
- The are boards that doesn't provide board.h nor periph_conf.h (e.g wsn_430_v1_*)

Any suggestions?

jia200x commented:
(Although it's desirable to write plain MD code instead of commented, I stillfeel it's a cleaner approach to define everything in the doc file)

aabadie commented:
I propose that we stick to the actual approach (e.g. defining board groups and global documentation in doc.txt) and we start thinking to a solution with full markdown files. Otherwise, this PR might never ends and it blocks new board support/changes (see #8649 and related).

jia200x commented:
>I propose that we stick to the actual approach (e.g. defining board groups and global documentation in doc.txt) and we start thinking to a solution with full markdown files. Otherwise, this PR might never ends and it blocks new board support/changes (see #8649 and related).

Is it fine for you @kaspar030 ?

kaspar030 commented:
> Is it fine for you @kaspar030 ?

Not really. Moving our nice markdown doxygen into ```/** * */``` somehow hurts.

How about keeping the actual markdown in its own file, and including into doxygen using ```@include doc.md```?

jia200x commented:
>How about keeping the actual markdown in its own file, and including into doxygen using @include doc.md?

Tried `@include` and `@includedoc` from doc.txt (adding doc.md to EXAMPLE_PATH). The boards are not listed.
Any other idea?

aabadie commented:
> Moving our nice markdown doxygen into /** * */ somehow hurts.

It depends, it is not meant to be edited by user but rather being read from the generated html. Personnally, I can leave with that.

kaspar030 commented:
> Tried @include and @includedoc from doc.txt (adding doc.md to EXAMPLE_PATH). The boards are not listed.

You'll need to add ```../..``` to ```EXAMPLE_PATH``` (as "make doc" gets executed in doc/doxygen), then ```@includedoc boards/..../xxxx.md```. Maybe add "*.md" to ```EXAMPLE_PATTERNS```. But then the markdown is not parsed anymore.

> Any other idea?

No. ;(

jia200x commented:
@kaspar030 I gave another try with `@subpage`, but still get the same results. 
I tend to be with @aabadie on this. The documentation is not meant to be edited by end users and another entry in the Boards group (the "doc" entry) displays weird to end users IMO. 
Also, we have been doing `/** * */` before (e.g [sock documentation](https://github.com/RIOT-OS/RIOT/blob/c02ee081d9580f6533c531518a98be9f58878cb8/sys/include/net/sock/ip.h)).

Opinions?

kaspar030 commented:
> Opinions?

IMO this shows that doxygen is not very good at non-API documentation.
Honestly, while I appreciate the work that was put into this move, it seems to me that the concept needs a little work.

1. now, hardware support documentation shows up as intro text of each board, in the API doc, under "Board specific definitions and implementations".

Before, "Modules" basically contained documentation of the logical part of RIOT. Now, hardware to-dos are mixed in there, mixed-in with board-specific defines. While that might be a sensible way to go, I do not see any discussion about it

2. there's a "doc.txt" that contains a C commen just so doxygen can parse it. inside the comment is markdown. This is confusing and hacky.

I'm somehow reluctant to ACK this, as while it solves "docs are in one place" and "help developers to keep end-user docs up-to-date", it worsens the overall structure of the documentation, both in the output and in its structure within the repository.

kaspar030 commented:
> I'm somehow reluctant to ACK this

I also don't want to block this PR. Maybe we can get some other opinions?

jia200x commented:
>Before, "Modules" basically contained documentation of the logical part of RIOT. Now, hardware to-dos are mixed in there, mixed-in with board-specific defines. While that might be a sensible way to go, I do not see any discussion about it

Basically the idea of this PR was just to move the existing Wiki entries to Doxygen. I fixed some broken links and tables, but the content is exactly the same. I do agree we should find a better way to expose stuff such as hardware TODOs, but that was not the original purpose of this PR.

>there's a "doc.txt" that contains a C commen just so doxygen can parse it. inside the comment is markdown. This is confusing and hacky.

This is a Doxygen limitation. I had the idea of reusing the board.h or periph_conf.h, but not all boards provide one. In any case, I don't see how it could be possible to (easily) use MD files without C style comments. If we assume the board documentation is not meant to be edited by the user (as @aabadie suggested), I think this is not the biggest issue.

>I'm somehow reluctant to ACK this, as while it solves "docs are in one place" and "help developers to keep end-user docs up-to-date", it worsens the overall structure of the documentation, both in the output and in its structure within the repository.

I tend to think this approach could help end users/developers in the short-mid term while we find another solution. Any comments?

jia200x commented:
@smlng maybe you are also interested in this PR

aabadie commented:
>  it worsens the overall structure of the documentation, both in the output and in its structure within the repository

@kaspar030 can you detail a bit more this ?

aabadie commented:
It's all green here, since @kaspar030 dismissed his review. Maybe it would be better to have someone else opinion as it's a quite large change with a big impact on documentation. @miri64 ?

miri64 commented on boards/acd52832/doc.txt:
```diff
@@ -0,0 +1,5 @@
+/**
+ * @defgroup    boards_acd52832 ACD52832
+ * @ingroup     boards
+ * @brief       Support for the aconno™ ACD52832
```
Are we okay with Unicode characters?

aabadie replied:
in #8379  this wasn't a problem, there's also one in cc2650stk doc

smlng replied:
I would remove such anyway (or replace by `(tm)` if necessary), I see super- and sub-scripts rendered badly more often than not.

jia200x replied:
I would also try to remove Unicode, same reason as @smlng says. In fact, it has happened to us in a RIOT related project

gebart commented:
I don't have time to read up on this thread right now, nor review anything. But I have a suggestion: Get rid of the leading ` * ` on each line inside the single comment block in each doc.txt. It will make editing easier, and it means that the file is the same as the discussed markdown file option, except for a `/**` on the first line and a `*/` on the last line.

miri64 commented:
> However, I also agree with the other, that when using markdown we should either have actual .md files or at least don't have all the lines starting with * like C-comments, because that makes future changes, extensions and additions rather painful.

This isn't possible with Doxygen. At least not in any of the current versions. *.md is for pages (see e.g. [this page](https://github.com/RIOT-OS/RIOT/blob/master/doc/doxygen/src/creating-an-application.md), and while we could go with pages, there are two downsides:

1. They can't be associated with groups and are on the top level of the table of contents (like ["Creating an application" is e.g.](http://doc.riot-os.org/creating-an-application.html)). This is why the TOC looked like it looked like in https://github.com/RIOT-OS/RIOT/pull/8516#discussion_r175132120
2. AFAIR (but I'm not sure) they should reside in a special directory (for us this is `doc/doxygen/src`).

I think @gebart's proposal is a nice compromise to make it at least a little easier on the eye for both readers and editors.

kaspar030 commented:
> However, I also agree with the other, that when using markdown we should either have actual .md files or at least don't have all the lines starting with * like C-comments, because that makes future changes, extensions and additions rather painful.

I could live with the latter.

We could consider creating doxygen files from markdown files in some way...

miri64 commented:
> We could consider creating doxygen files from markdown files in some way...

I don't know how... Doxygen seems to be hardwired to interpret files containing non-comments as either as (undocumented) code or pages. The only specific mention of "including markdown" in Doxygen's doc I can find is [this page](http://www.stack.nl/~dimitri/doxygen/manual/markdown.html#md_page_header) (which BTW is in its source [exactly as the proposed style](https://github.com/doxygen/doxygen/blob/master/doc/markdown.doc)).

Just to be clear, what I mean by "pages": The file is read as though it belongs to a [`@page`](http://www.stack.nl/~dimitri/doxygen/manual/commands.html#cmdpage) command within the documentation.

tcschmidt commented:
@jia200x ping?

jia200x commented:
As @miri64 pointed out, having *.md files embedded with the Modules is not a Doxygen features, and have some downsides (weird TOC, some hacks, etc). 

I really like @gebart proposal. I will do that and see how works.

tcschmidt commented:
> I really like @gebart proposal. I will do that and see how works.

And? Does it work well?

jia200x commented:
> And? Does it work well?

I will push something today and then ask for consensus :)

jia200x commented:
@tcschmidt just pushed the new changes. Doxygen seems to be more strict without the * characters (e.g now it's necessary to have `\n` after \`\`\`, otherwise it fails to compile the documentation)

The rest works fine.

jia200x commented:
@miri64 @tcschmidt @aabadie @kaspar030 I addressed @gebart proposition and everything is working. As stated before, the entries are not perfect (still updated information from the wiki), but I propose to address those stuff in separates PRs and slowly adopt the new documentation.

Are you OK with merging this PR?

tcschmidt commented:
@miri64 @aabadie @kaspar030 anybody ready to check and ack?

jia200x commented:
could someone else check this one? :)

tcschmidt commented:
@smlng any new insights? Since @aabadie is gone, maybe you can push the review.

smlng commented on boards/nucleo-f031k6/doc.txt:
```diff
@@ -0,0 +1,2 @@
+/**
```
why is this empty? Shouldn't it at least have the `@defgroup`?

jia200x replied:
yes, true. I will fix it

smlng commented on boards/nucleo-f042k6/doc.txt:
```diff
@@ -0,0 +1,2 @@
+/**
```
same here, see above

jia200x commented:
>could you please also provide the script you uses, may be put into dist/tools for future reference and usage.
The problem is those scripts are just hacky bash instructions that were adapted for each case. There's no script to consistently generate doc.txt from header files :(

jia200x commented:
@smlng I already pushed the requested changes. There are still some boards that don't have the new format, but I think it's better to open a PR as soon as this one gets merged (otherwise this will be in the PR list forever)

jia200x commented:
@miri64 @smlng are you ok with the actual state?

smlng commented:
side note: images for samd21 and saml21 are not working anymore (broken in wiki, too) because microchip overtook atmel. I fixed in the wiki, so you might want to update this here too?

jia200x commented:
> side note: images for samd21 and saml21 are not working anymore (broken in wiki, too) because microchip overtook atmel. I fixed in the wiki, so you might want to update this here too?

Good catch. Thanks

Sure, no problem

jcarrano commented:
>  images for samd21 and saml21 are not working anymore

We should not be deep-linking content.

smlng commented:
@jcarrano I agree, but currently we have no common strategy where to put or link images to/from - I guess copyright might be an issue, so putting them into the repo is not an option I guess. 

Hence, for now it was the easiest (and quickest) to fix the links by replacing the URLs.

jia200x commented:
@smlng alles gut. Fixed SAML21 and SAMD21 entries.

jia200x commented:
ping @miri64

aabadie commented:
Finally !!!

smlng commented:
Nice works, congrats @jia200x

jia200x commented:
Thank you to all people involved! I will put deprecation warnings in the wiki's board entries

miri64 commented:
> I will put deprecation warnings in the wiki's board entries

But please with link to knew place ;-) (I think http://doc.riot-os.org/group__boards.html should suffice).
